pylint/checkers/imports.py

```diff
@@ -826,6 +828,12 @@ def _add_imported_module(self, node, importedmodname):
         except ImportError:
             pass
 
+        in_type_checking_block = (
```

Comment:
Regarding my comment in another MR, maybe we should create two functions in ``astroid.If`` : ``is_typing_guard`` and ``is_sys_guard`` ?